File: src/immersed.dive.shop.model/Person.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace immersed.dive.shop.model
{
    public class Person : IEntity
    {
        public Person()
        {
            Id = Guid.NewGuid();
            Events = new List<EventParticipant>();
        }

        public Guid Id { get; set; }
        public string FullName { get; set; }
        public string FamilyName { get; set; }
        public string Name { get; set; }
        public string OtherNames { get; set; }
        public string KnownAs { get; set; }
        public DateTime DateOfBirth { get; set; }
        public Sex Sex { get; set; }
        public string IdentifiesAs { get; set; }

        public List<EventParticipant> Events { get; set; }

        public DateTime DateCreated { get; set; }
        public DateTime LastUpdated { get; set; }
        public bool Live { get; set; }
    }
}

File: the-client-app/src/app/models/person.ts

export interface Person {
    id?: string;    
    fullName?: string;    
    familyName?: string;    
    name?: string;    
    otherNames?: string;    
    knownAs?: string;    
    dateOfBirth?: Date
    sex?: string;    
    identifiesAs?: string;    
    dateCreated?: Date;
    lastUpdated?: Date;
    live?:boolean;      
}

export class Person implements Person {
    constructor(person?:Person) {
        if(person) {
            this.id = person.id;

            this.fullName = person.fullName;
            this.familyName = person.familyName;
            this.name = person.name;
            this.otherNames = person.otherNames;
            this.knownAs = person.knownAs;
            this.dateOfBirth = person.dateOfBirth;
            this.sex = person.sex;
            this.identifiesAs = person.identifiesAs;
            
            this.dateCreated = person.dateCreated;
            this.lastUpdated = person.lastUpdated;
            this.live = person.live;
        }
    }
}
